autoreject/autoreject.py

```diff
@@ -377,15 +376,15 @@ def compute_thresholds(epochs, method='bayesian_optimization',
             data = np.concatenate((epochs.get_data(),
                                    epochs_interp.get_data()), axis=0)
             y = np.r_[np.zeros((n_epochs, )), np.ones((n_epochs, ))]
-        cv = StratifiedShuffleSplit(y, n_iter=10, test_size=0.2,
+        cv = StratifiedShuffleSplit(n_splits=10, test_size=0.2,
                                     random_state=random_state)
 
         ch_names = epochs.ch_names
 
         my_thresh = delayed(_compute_thresh)
         verbose = 51 if verbose is not False else 0  # send output to stdout
         threshes = Parallel(n_jobs=n_jobs, verbose=verbose)(
-            my_thresh(data[:, pick], cv=cv, method=method,
+            my_thresh(data[:, pick], cv=cv, method=method, y=y,
```

Comment:
I don't understand why you need to pass y around. For me it's always None no?

Reply:
No, it's not always `None`. When we have augmented data, we want to do stratified cross validation so that there are roughly equal number of original trials and augmented trials in each fold

Reply:
hum but you use KFold by default which is not stratified. You should use : StratifiedKFold

also it seems you want to use something closer to the groups parameter so you can use GroupKFold. It will allow you to assert that an augmented trial is not part of the test set if the orig trial is in the train set.

Reply:
we do use stratified shuffle split here: https://github.com/autoreject/autoreject/blob/9b05a796eb5e80649de1e988b104a99d53bddd4c/autoreject/autoreject.py#L380

However, it is used only for estimating thresholds, not for estimating `rho` and `kappa`. It is an old open PR but remains to be tested: https://github.com/autoreject/autoreject/pull/28/files.

The goal of this PR is to keep autoreject the same but remove sklearn deprecations